src/Datastore/DatastoreClient.php

```diff
@@ -449,30 +451,89 @@ public function allocateIds(array $keys, array $options = [])
     }
 
     /**
-     * Create a Transaction
+     * Create a Transaction.
      *
      * Example:
      * ```
      * $transaction = $datastore->transaction();
      * ```
      *
      * @see https://cloud.google.com/datastore/docs/concepts/transactions Datastore Transactions
+     * @see https://cloud.google.com/datastore/docs/reference/rest/v1/projects/beginTransaction beginTransaction
      *
-     * @param array $options [optional] Configuration options.
+     * @codingStandardsIgnoreStart
+     * @param array $options {
+     *     Configuration options.
+     *
+     *      @type string $previous If the transaction is being restarted,
+     *           the previous transaction ID.
+     *     @type array $transactionOptions Any other transaction configuration,
+     *           besides `previous`. See
+     *           [ReadWrite](https://cloud.google.com/datastore/docs/reference/rest/v1/projects/beginTransaction#ReadWrite).
+     * }
      * @return Transaction
+     * @codingStandardsIgnoreEnd
      */
     public function transaction(array $options = [])
     {
-        $res = $this->connection->beginTransaction($options + [
-            'projectId' => $this->projectId
-        ]);
+        $transactionOptions = $this->pluck('transactionOptions', $options, false) ?: [];
+        $previous = $this->pluck('previous', $options, false);
+        if ($previous) {
+            $transactionOptions['previousTransaction'] = $previous;
+        }
+
+        $res = $this->connection->beginTransaction([
+            'projectId' => $this->projectId,
+            'transactionOptions' => [
+                // if empty, force request to encode as {} rather than [].
+                'readWrite' => $transactionOptions ?: (object) []
+            ]
+        ] + $options);
```

Comment:
This may cause issues where users cannot override `transactionOptions` as this doesn't perform a recursive merge.